roles/foreman_maintain_packages/tasks/main.yaml

```diff
@@ -1,3 +1,7 @@
 ---
+- name: Gather the package facts
+  package_facts:
+    manager: auto
 - name: 'Allow Packages to be Installed'
-  command: "foreman-maintain packages unlock -y"  
\ No newline at end of file
+  command: "foreman-maintain packages unlock -y" 
+  when: "'rubygem-foreman_maintain' in ansible_facts.packages"
```

Comment:
Isn't it sufficient to just check if `foreman-maintain` is in `$PATH`? In the installer we use this:
https://github.com/theforeman/foreman-installer/blob/3cbb7bd7201654a491a927cbb3d6867f50523c99/hooks/boot/09-version_locking.rb#L1-L3

That was contributed by the author of the locking feature as a way to check if locking is supported.

The reason is that foreman-maintain may be installed, but not support locking. For example, if we do start supporting Debian then it may not support the feature.

Reply:
There is no easy way to check if something is in `$PATH` in Ansible.

So yeah, calling it with `-h` and only proceed if that doesn't error is a possibility, but I think we should cross that bridge when we get FM on Debian without locking.

Reply:
One thing that I could do is create another 2 steps, one for RHEL Family and another one for Debian family, and do nothing when it's detect that is on Debian.

Reply:
Given the package will have a different name on Debian anyways, I think this is rather moot ;)